routers/repo/issue.go

```diff
@@ -1055,6 +1055,30 @@ func UpdateIssueTitle(ctx *context.Context) {
 	})
 }
 
+// UpdateIssueRef change issue's ref (branch)
+func UpdateIssueRef(ctx *context.Context) {
+	issue := GetActionIssue(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	if !ctx.IsSigned || (!issue.IsPoster(ctx.User.ID) && !ctx.Repo.CanWriteIssuesOrPulls(issue.IsPull)) {
```

Comment:
@vedranMv This was an oversight from my part. Although the UI will not let the user change the branch on pull requests, please add a check here too so PRs can't be changed by using the API directly.

Also, I think we should also check that the issue is not closed (that check should also be added to the UI).

Reply:
If user try to change a branch of a PR, we should return an error.

Reply:
Sorry for the delay. Something like 2bd8e64 ?